hw/ip/csrng/doc/_index.md

```diff
@@ -0,0 +1,89 @@
+- Compliant with NIST and BSI recommendations for random number generation.
+- In deterministic mode, meets the requirements given in AIS31 for a DRG.3 class deterministic random number generator (DRNG) meaning it provides Forward Secrecy and Enhanced Backward Secrecy.
+- Utilizes the CTR_DRBG construction specified in NIST SP 800-90A, qualifying it as a NIST-approved DRBG (deterministic random bit generator).
+    - Operates at 256 bit security strength.
+- Support for multiple separate CSRNG instances per IP block.
+    - Each instance has its own internal state, control, reseed counters and IO pins.
+    - The number of RNG instances is set via a module parameter.
+- Software access to a dedicated RNG instance.
+    - One instance, Instance 0, is always accessible from the bus through device registers,
+    - All other instances route to other hardware peripherals (e.g. the key manager, obfuscation engines etc.) and in normal operation these other instances are inaccessible from software.
+    - The IP may be configured to support "debug mode" wherein all instances can be accessed by software.
+      For security reasons this mode may be permanently disabled using one-time programmable (OTP) memory..
+- The IP interfaces with external entropy sources to obtain any required non-determinstic seed material (entropy) and nonces.
+    - Requires an external entropy source which is compliant with NIST SP 800-90B, and which also satisfies the requirements for a PTG.2 class physical non-deterministic random number generator as defined in AIS31.
+    - Dedicated hardware interface with external entropy source satisfies requirements for `get_entropy_input()` interface as defined in SP 800-90A.
+    - When needed, utilizes the `Block_Cipher_df` derivation function (as defined in SP 800-90A) for assembling seed material.
+      This allows the use of entropy sources which are not full-entropy (less than one bit of entropy per bit generated).
+- Also supports the optional use of personalization strings or other application inputs (e.g. OTP memory values) during instantiation.
+- Assuming a continuously-live entropy source, each instance can also optionally be used as a non-determinstic TRNG (true random number generator, also called a non-deterministic random bit generator or NRBG in SP 800-90C).
+    - In this mode, an instance also meets the requirements laid out for a PTG.3 class RNG, the strongest class laid out in AIS31.
+    - Implementation follows the NRBG "Oversampling Construction" approved by SP 800-90C, to meet both CC and FIPS TRNG constructions.
+- In addition to the approved DRNG mode, any instance can also operate in "Fully Deterministic mode", meaning the seed depends entirely on application inputs or personalization strings.
+    - This provides an approved means of key-derivation in FIPS 140-2.
```

Comment:
If you have it handy, can we be a little more precise with the reference to FIPS 140-2, i.e. what section says this means is approved?

Reply:
Here it is: 
`This provides an approved means of seed construction in FIPS 140-2 as described in the [FIPS 140-2 Implementation Guidance](https://csrc.nist.gov/csrc/media/projects/cryptographic-module-validation-program/documents/fips140-2/fips1402ig.pdf), section 7.14, resolution point 2(a).`
Note I've generalized this here to seed generation as key generation is not specifically mentioned there, but we do like this mechanism specifically for key generation.